contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,368 @@
+# provided that the above copyright, this permission notice and text
+# this comment, and the disclaimer below appear in all of the following:
+#
+#       supporting documentation
+#       source copies
+#       source works derived from this source
+#       binaries derived from this source or from derived source
+#
+# Landon Curt Noll, Martin Kontsek DISCLAIMS ALL WARRANTIES WITH REGARD
+# TO THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+# AND FITNESS. IN NO EVENT SHALL Landon Curt Noll AND Martin Kontsek BE
+# LIABLE FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+
+####################################################################
+#                                                                  #
+#                  Libfabric RPM build script                      #
+#                                                                  #
+####################################################################
+
+####################
+# initialized values
+####################
+export specfile="libfabric.spec"
+export rpmbuilddir="$PWD/rpmbuild"
+export prefix="/usr"
+export noop=""
+export install_in_opt=""
+export create_modulefile=""
+export unpack_spec=""
+export verbose=""
+
+##################################
+# If you need custom configure and
+# rpmbuild options, put them here
+##################################
+export configure_options=""
+export rpmbuild_options=""
+
+###################
+# usage information
+###################
+usage="Usage: $0 [-i provider_name] [-e provider_name]
+       [-n] [-o] [-m] [-d] [-u] [-s] [-v] [-h] tarball
+
+ Provider options:
+
+  -i    provider_name
+    include 'provider_name' provider support to the build
+
+  -e    provider_name
+    exclude 'provider_name' provider support from the build
+
+ General options:
+  -n         no op, do nothing
+               useful with -v option
+
+  -o         install under /opt/libfabric/_VERSION_
+               {default: install under /usr/ }
+
+  -m         create modulefile
+               {default: don't create modulefile}
+
+  -d         build with Debugging support
+              {default: without debugging support}
+
+  -u         build with usNIC support only (do NOT combine with other build options)
+               {default: build with support for usNIC, Infiniband, Sockets, etc.}
+
+  -s         try to unpack libfabric.spec file from tarball
+               {default: look for libfabric.spec file in current directory}
+
+  -v         be verbose
+
+  -h         print this message and exit
+
+  tarball    path to Libfabric source tarball
+  "
+
+############
+# parse args
+############
+export arguments="$@"
+while getopts nomi:e:udsvh flag; do
+    case "$flag" in
+      n) noop="true"
```

Comment:
The reason for using [[ instead test command was, that we can put variables inside " to make script more bulletproof. The other reason is that I am used to C syntax, and this feel more familiar. 
The true instead of 1 was used to improve readability of the script, as you can directly see that it stands for boolean.